### PR TITLE
ref(ts): Refactor Feature component to TypeScript

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/acl/feature.tsx [hunk 1]
```diff
@@ -146,7 +167,7 @@ class Feature extends React.Component {
     // Override the renderDisabled function with a hook store function if there
     // is one registered for the feature.
     if (hookName) {
-      const hooks = HookStore.get(`feature-disabled:${hookName}`);
+      const hooks: [Function] = HookStore.get(`feature-disabled:${hookName}`);
```
Comment (dashed) [inline src/sentry/static/sentry/app/components/acl/feature.tsx]: Heads up, `[Function]` is a tuple type. 

I think you may want `Function[]` or `Array<Function>`.

**Review by dashed — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/components/acl/feature.tsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: There's only one element in the list with a type of Function, isn't tuple correct in this case?

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: Actually, not 100% sure what hooks is in this case, but in the code below we only ever use the first element of the array.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: Yeah it should be a list, it just pulls the first one off.

File: src/sentry/static/sentry/app/components/acl/feature.tsx [hunk 2]
```diff
@@ -200,4 +225,4 @@ class FeatureContainer extends React.Component {
   }
 }
 
-export default withConfig(FeatureContainer);
+export default withConfig(withOrganization(withProject(FeatureContainer)));
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/acl/feature.tsx]: Why wrap this, we are already pulling out the org and project from the context in the `FeatureContainer`?

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: Ah, thanks. Meant to delete that and use the HoC

File: src/sentry/static/sentry/app/components/acl/feature.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/acl/feature.tsx]: I based my feedback on the types from https://github.com/getsentry/sentry/pull/14160



-----

Looks like it's resolved.  👍

**Review by dashed — APPROVED** [review]
Looks good to me.

**Pull request merged**